### PR TITLE
optimalise highlight in title of searchresult

### PR DESCRIPTION
**Pull request by StuyckRuben**
https://kanselarij.atlassian.net/browse/KAS-3965

### CODE REVIEW

File: app/routes/search/agendaitems.js [hunk 1]
```diff
@@ -62,6 +62,10 @@ export default class AgendaitemSearchRoute extends Route {
     if (entry.title && Array.isArray(entry.title)) {
       entry.title = entry.title.join('');
     }
+    if (agendaitem.highlight.shortTitle[0]?.split(" ").length < agendaitem.attributes.shortTitle?.split(" ").length) {
```
Comment (tdn) [inline app/routes/search/agendaitems.js]: ~~This doesn't do the proper check, as the `agendaitem.highlight.shortTitle[0]` will still include the ems. To be discussed~~

Comment (tdn) [reply app/routes/search/agendaitems.js]: oh wait, nevermind. It's the length of the split array that's checked here, which will work

Comment (tdn) [reply app/routes/search/agendaitems.js]: test is likely failing here when `agendaitem.highlight` is `null`, and/or when `agendaitem.highlight.shortTitle` is falsy (undefined or null). 

Pretty sure `agendaitem.highlight.shortTitle` is always an Array when it's not null, but can't hurt to check either.

@StuyckRuben remember to not always go by the happy flow. Always try to break your own code ;-)

**Review by tdn — CHANGES_REQUESTED** [review]

**Review by ValenberghsSven — COMMENTED** [review]
do we want to check all highlights?
Now we only check the first one.
So not all hits are highlighted.


image

The data: 
image

**Comment by ValenberghsSven** [comment]
Other than the "issue" mentioned above it seems fine.
Searching on other fields (like mandatee) without highlights in titles works (no errors)

File: app/routes/search/agendaitems.js [hunk 2]
```diff
@@ -62,6 +62,10 @@ export default class AgendaitemSearchRoute extends Route {
     if (entry.title && Array.isArray(entry.title)) {
       entry.title = entry.title.join(' ');
     }
+    if ((agendaitem.highlight && agendaitem.highlight.shortTitle) && agendaitem.highlight.shortTitle[0]?.split(" ").length < agendaitem.attributes.shortTitle?.split(" ").length) {
+      const highlightShortTitleEmsRemoved = agendaitem.highlight.shortTitle[0].replace(/<em>|<\/em>/g, "");
+      entry.shortTitle = agendaitem.attributes.shortTitle.replace(highlightShortTitleEmsRemoved, agendaitem.highlight.shortTitle[0]);
```
Comment (tdn) [inline app/routes/search/agendaitems.js]: change this to `replaceAll` and Sven's comment should be addressed

**Review by tdn — CHANGES_REQUESTED** [review]

File: app/routes/search/agendaitems.js (hunk 2 above)
Comment (tdn) [reply app/routes/search/agendaitems.js]: correction: this should be in a for loop, since we're replacing the whole highlight snippet

**Review by tdn — APPROVED** [review]

**Pull request merged**